File: scala-plugin-netlogo-sbt/src/MyExtension.scala
package org.test

import org.nlogo.{ agent, api, nvm }
import api.Syntax._
import api.ScalaConversions._  // implicits

class MyExtension extends api.DefaultClassManager {

  def load(manager: api.PrimitiveManager) {
    manager.addPrimitive("print-message", new MyMessage())
    manager.addPrimitive("get-first-character", new ComputeMean())
    manager.addPrimitive("build-a-random-list", new BuildRandomList())
  }

}

File: java-plugin-netlogo-maven/src/main/java/org/test/MyExtension.java
package org.test;

import org.nlogo.api.DefaultClassManager;
import org.nlogo.api.ExtensionException;
import org.nlogo.api.PrimitiveManager;

public class MyExtension extends DefaultClassManager {
  public void load (PrimitiveManager primitiveManager) throws ExtensionException {
      primitiveManager.addPrimitive("print-message", new MyMessage());
      primitiveManager.addPrimitive("get-mean", new ComputeMean());
      primitiveManager.addPrimitive("build-a-random-list", new BuildRandomList());
  }

}
